contrib/nydusd-http-server/src/main.rs

```diff
@@ -0,0 +1,236 @@
+                _host: host,
+                range: rangestr,
+                _authstr: authstr,
+            })
+        }
+    }
+
+    pub struct Layerinfo(NamedFile, String);
+    impl<'r> Responder<'r, 'static> for Layerinfo {
+        fn respond_to(self, _req: &'r Request<'_>) -> response::Result<'static> {
+            Response::build_from(self.0.respond_to(_req)?)
+                .raw_header("Docker-Content-Digest", self.1)
+                .ok()
+        }
+    }
+
+    #[rocket::head("/<_namespace>/<_repo>/blobs/<digest>")]
+    pub async fn layer_exist(
+        _namespace: PathBuf,
+        _repo: PathBuf,
+        digest: String,
+    ) -> Result<Option<Layerinfo>, Status> {
+        let args: Vec<String> = env::args().collect();
+        let mut path = Path::new(&args[1]).join("blobs");
+        let dis = &digest[7..];
+        if path.is_dir() {
+            path.push(&dis);
+        }
+        Ok(NamedFile::open(path)
+            .await
+            .ok()
+            .map(|nf| Layerinfo(nf, dis.to_string())))
+    }
+
+    /* fetch blob response
+     * NamedFile: blob data
+     * String: Docker-Content-Digest
+     */
+    pub struct CachedFile(NamedFile, String);
+
+    impl<'r> Responder<'r, 'static> for CachedFile {
+        fn respond_to(self, req: &'r Request<'_>) -> response::Result<'static> {
+            Response::build_from(self.0.respond_to(req)?)
+                .raw_header("Docker-Content-Digest", self.1)
+                .raw_header("Content-Type", "application/octet-stream")
+                .ok()
+        }
+    }
+
+    /* fetch blob part response(stream)
+     * path: path of blob
+     * dis: Docker-Content-Digest
+     * start & end: "Content-Range: bytes <start>-<end>/<size>"
+     */
+    pub struct RangeStream {
+        path: PathBuf,
+        dis: String,
+        start: u64,
+        end: u64,
+    }
+
+    impl<'r> Responder<'r, 'static> for RangeStream {
+        fn respond_to(self, _req: &'r Request<'_>) -> response::Result<'static> {
+            let rangestring = format!(
+                "bytes {}-{}/{}",
+                self.start,
+                self.end,
+                self.end - self.start
+            );
+            let size = self.end - self.start + 1;
+            let mut file = match fs::File::open(self.path) {
+                Ok(f) => f,
+                Err(_err) => {
+                    return Err(Status::NotFound);
+                }
+            };
+            match file.seek(io::SeekFrom::Start(self.start)) {
+                Ok(file) => file,
+                Err(_err) => {
+                    return Err(Status::InternalServerError);
+                }
+            };
+            Response::build()
+                .streamed_body(ReaderStream! {
+                const BUFSIZE : usize=1048576;  //bufsize:2M
+                let mut buf = [0; BUFSIZE];
+                let mut readed = 0;
+                    while readed < size
```

Comment:
The indent before `while`  here looks strange . Did you try to format your docments?